From f75e5c3d84ee5a67c6ebc5aeb5c708a27d934ae0 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Mon, 28 Apr 2014 22:13:52 +0200
Subject: [PATCH 16/23] MINOR: http: remove the now unused loop over "block"
 rules

This ruleset is now always empty, simply remove it.
---
 src/proto_http.c | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/src/proto_http.c b/src/proto_http.c
index 14b0835..3cfa8e2 100644
--- a/src/proto_http.c
+++ b/src/proto_http.c
@@ -3813,7 +3813,6 @@ int http_process_req_common(struct session *s, struct channel *req, int an_bit,
 {
 	struct http_txn *txn = &s->txn;
 	struct http_msg *msg = &txn->req;
-	struct acl_cond *cond;
 	struct http_req_rule *http_req_last_rule = NULL;
 	struct redirect_rule *rule;
 	struct cond_wordlist *wl;
@@ -3837,18 +3836,6 @@ int http_process_req_common(struct session *s, struct channel *req, int an_bit,
 	/* just in case we have some per-backend tracking */
 	session_inc_be_http_req_ctr(s);
 
-	/* first check whether we have some ACLs set to block this request */
-	list_for_each_entry(cond, &px->block_rules, list) {
-		int ret = acl_exec_cond(cond, px, s, txn, SMP_OPT_DIR_REQ|SMP_OPT_FINAL);
-
-		ret = acl_pass(ret);
-		if (cond->pol == ACL_COND_UNLESS)
-			ret = !ret;
-
-		if (ret)
-			goto deny;
-	}
-
 	/* evaluate http-request rules */
 	http_req_last_rule = http_req_get_intercept_rule(px, &px->http_req_rules, s, txn);
 
-- 
1.8.3.2

